python/lsst/cp/pipe/ptc.py

```diff
@@ -365,10 +376,102 @@ def runDataRef(self, dataRef, visitPairs):
+                        ("LINEARIZESQUARED", 'linearizeSquared')]:
+
+            linType = linPair[0]
+            dataType = linPair[1]
+            if linType == "LOOKUPTABLE":
+                tableArray = lookupTableArray
+            else:
+                tableArray = None
+
+            linearizer = self.buildLinearizerObject(dataset, detName, detNum, detector,
+                                                    instruName=self.config.instrumentName,
+                                                    linearizerType=linType, tableArray=tableArray,
+                                                    log=self.log)
+            butler.put(linearizer.toDict(), datasetType=dataType, dataId={'detector': detNum,
+                       'detectorName': detName, 'calibDate': now.strftime("%Y-%m-%d")})
+
         self.log.info('Finished measuring PTC for in detector %s' % detNum)
 
         return pipeBase.Struct(exitStatus=0)
 
+    def buildLinearizerObject(self, dataset, detName, detNum, detector, instruName='', linearizerType='',
+                              tableArray=None, filt='NONE', log=None):
+        """Build linearizer object to persist.
+
+        Parameters
+        ----------
+        dataset : `lsst.cp.pipe.ptc.PhotonTransferCurveDataset`
+            The dataset containing the means, variances and exposure times
+        detName : `srt`
+            Detector name
+        detNum : `int`
+            Detector number
+        detector : `lsst.afw.cameraGeom.detector.detector.Detector`
+            Detector object
+        instruName : `str`, optional
+            Instrument name
+        linearizerType : `str`, optional
+            'LOOKUPTABLE' or 'LINEARIZESQUARED'
+        tableArray : `np.array`, optional
+            Look-up table array with size rows=nAmps and columns=ADU values
+        filt : filter, `str`, optional
+        log : `lsst.log.Log`, optional
+            Logger to handle messages
+
+        Returns
+        -------
+        linearizer : `lsst.ip.isr.Linearizer`
+            Linearizer object
+        """
+        if linearizerType == 'LOOKUPTABLE':
+            if tableArray is not None:
+                linearizer = Linearizer(table=tableArray, log=log)
+            else:
+                raise RuntimeError("tableArray must be provided when creating a LookupTable linearizer")
+        elif linearizerType in ['LINEARIZESQUARED', 'LINEARIZEPOLYNOMIAL']:
+            linearizer = Linearizer(log=log)
+        else:
+            raise RuntimeError("Enter supporter linearizerType to build a Linearizer object: " +
+                               "'LOOKUPTABLE', 'LINEARIZESQUARED', or 'LINEARIZEPOLYNOMIAL'")
+        linearizer.setMetadata()
+        for i, amp in enumerate(detector.getAmplifiers()):
+            ampName = amp.getName()
+            if linearizerType == 'LOOKUPTABLE':
+                linearizer.linearityCoeffs[ampName] = [i, 0]
+                linearizer.linearityType[ampName] = "LookupTable"
```

Comment:
A way to remove the linearity type from the if clauses would be to define the mapping in a dict:
```python
linearityType = {"LOOKUPTABLE": "LookupTable", "LINEARIZESQUARED": "Squared"}
for ... in ...:
    linearizer.linearityType[ampName] = linearityType[linearizerType]
```

Saying that, why do you need to tell the Linearizer object what each amp is when it seems to be a constant? Am I seeing a special case here or is it always true that every amp has the same type? Could you instead have Linearizer subclasses that have their type defined for them like @SimonKrughoff did for the Curve class?  Maybe @czwa can tell me.  Of course if each amp can have different types then that would explain this but the code here doesn't allow that.

Reply:
Having this be amplifier based instead of detector (or camera) based is a result of directly mapping the cameraGeom linearity parameters to the new Linearizer.  I also didn't want to prevent different amplifiers from having different correction types (because electronics can always be difficult in unique ways).

Reply:
So this means that Linearizer can support per-amp but this cp_pipe task can't. Can this be mentioned somewhere in the task docstring?

Reply:
I left the "if" clauses because there are other things specific to each linearizer type that are being set. I added clarification regarding the linearizers in the doc-string of "MeasurePhotonTransferCurveTask".